### DISCUSSION TITLE
How to cancel menu on touch screen?

### DISCUSSION THREAD

**Discussion by swsnr** [Schneegans/Fly-Pie#260] [Fly-Pie Q&A]
Hello,

how can I cancel open menus on a touch screen without activating any item?

I know that I can right-click with a mouse, but that's rather inconvenient on a touchscreen 😇

**Comment by swsnr** [comment]
Okay, I just found out that I can apparently cancel menus with a long press anywhere on the screen while the menu is open 😬 

Not sure whether this is intentional… I found this a bit surprising: I only discovered this because I frantically tried to close the menu by touching the touch button again, and accidentally pressed too long.

I think the whole functionality around touch buttons would benefit from more discoverability.  I also had to consult https://github.com/Schneegans/Fly-Pie/discussions/186 to find out how to move a touch button.  Perhaps a dedicated tutorial for touch buttons?  Or a small help text beneath the toggle in the menu editor?

**Comment by claudiusraphael** [comment]
If the menu was opened via touch providing a cancel-button or alternatively a check for touching outside of the menu-area to trigger a cancel would be nice.

**Comment by Schneegans** [comment]
"outside of the menu-area" is a bit difficult, as there is no "outside". It's the strength of a pie menu that you can click anywhere on the screen to select items. However, a single click on the center currently does nothing. Maybe we could use this for cancelling?

**Reply by claudiusraphael** [reply]
Sounds good, had that thought as well. On that note i wondered if this space could be used also as a separate interactive space for (in the most novel case animated) interaction-info-icons. I do not have any insight in the code, its structure, what is possible or even desired - my only experience regarding pie-menus was programming some in js/as for html and flash pages in .. well before the dinosaurs sotosay, lol. What i originally meant by outside of the pie-menu was implying an invisible ring-space/donut around the pie-menu, if that makes sense?

**Reply by Schneegans** [reply]
I created #277 accordingly.
